### PR TITLE
NONPR-3350: Add notable event mdc context to retrieve and download requests

### PR DESCRIPTION
**Pull request by AIbrahimQA**
…quests

### CODE REVIEW

File: app/uk/gov/hmrc/nrsretrievalfrontend/views/search_page.scala.html [hunk 1]
```diff
@@ -41,6 +41,8 @@
     request.notableEvent
 }
 
+<script type="text/javascript"> var notableEvent = { name: "@notableEvent.name" };</script>
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/nrsretrievalfrontend/views/search_page.scala.html:44]: does this need to be a var? 

surely if you have a hidden input html field with the value you can pull it?

File: app/uk/gov/hmrc/nrsretrievalfrontend/controllers/SearchController.scala [hunk 2]
```diff
@@ -58,7 +58,7 @@ class SearchController @Inject()(
 
   private val action: String => ActionBuilder[NotableEventRequest, AnyContent] = {
     notableEventType =>
-      MDC.put("notableEvent", notableEventType)
+      MDC.put("notable_event", notableEventType)
```
[the remark this answers is not in the document]
Comment (AIbrahimQA) [reply app/uk/gov/hmrc/nrsretrievalfrontend/controllers/SearchController.scala:61]: When using notableEvent it comes out notableevent in Kibana.

[the remark this answers is not in the document]
Comment (DanielConnelly) [reply app/uk/gov/hmrc/nrsretrievalfrontend/controllers/SearchController.scala:61]: this is due to a rule in HMRC kibana to lowercase everything, therefore _ prefered over camel case.

File: app/uk/gov/hmrc/nrsretrievalfrontend/views/search_page.scala.html (hunk 1 above)
Comment (AIbrahimQA) [reply app/uk/gov/hmrc/nrsretrievalfrontend/views/search_page.scala.html]: Updated

File: app/uk/gov/hmrc/nrsretrievalfrontend/views/components/SearchResultPanel.scala.html [hunk 3]
```diff
@@ -57,7 +59,7 @@
         data-index="@{index}"
         role="button"
         onclick="setStatus(@index, 'Downloaded')"
-        href="@{s"/nrs-retrieval/download/${searchResult.vaultId}/${searchResult.archiveId}"}"
+        href="@{s"/nrs-retrieval/download/${request.notableEvent.name}/${searchResult.vaultId}/${searchResult.archiveId}"}"
```
Comment (DanielConnelly) [inline app/uk/gov/hmrc/nrsretrievalfrontend/views/components/SearchResultPanel.scala.html:62]: could we use the reverse router here please?

File: app/assets/javascripts/nrs.js [hunk 4]
```diff
@@ -128,7 +129,7 @@ function doRetrieve(index, vaultName, archiveId) {
   setStatus(index, STATUS_INCOMPLETE)
   const xmlhttp = http(index, vaultName, archiveId, timeout);
 
-  xmlhttp.open(GET, PATH + 'retrieve/' + vaultName + '/' + archiveId);
+  xmlhttp.open(GET, PATH + 'retrieve/' + notableEventType.name + '/' + vaultName + '/' + archiveId);
```
Comment (DanielConnelly) [inline app/assets/javascripts/nrs.js:132]: is name the right thing here?

Comment (DanielConnelly) [reply app/assets/javascripts/nrs.js:132]: please ignore, I was being an idiot.

**Review by DanielConnelly — APPROVED** [review]

**Pull request merged**